app/models/manageiq/providers/hawkular/datawarehouse_manager/event_catcher/runner.rb

```diff
@@ -0,0 +1,73 @@
+class ManageIQ::Providers::Hawkular::DatawarehouseManager::EventCatcher::Runner <
+  ManageIQ::Providers::BaseManager::EventCatcher::Runner
+
+  include ManageIQ::Providers::Hawkular::Common::EventCatcher::RunnerMixin
+
+  TAG_URL  = 'url'.freeze
+  TAG_TYPE = 'type'.freeze
+
+  def initialize(cfg = {})
+    super
+  end
+
+  def self.log_handle
+    $datawarehouse_log
+  end
+
+  def self.event_monitor_class
+    ManageIQ::Providers::Hawkular::DatawarehouseManager::EventCatcher::Stream
+  end
+
+  def self.find_target(tags)
+    target = case tags[TAG_TYPE].downcase
+             when "node" then OpenStruct.new(
+               :klass_name     => 'ContainerNode',
+               :find_key       => :name,
+               :value_tag_name => 'nodename',
+             )
+             else
+               $datawarehouse_log.error("unexpected target: [#{tags[TAG_TYPE]}]")
+             end
+    instance = nil
+    instance = target.klass_name.constantize.find_by(target.find_key => tags[target.value_tag_name]) if target
+    $datawarehouse_log.error("Could not find hawkular alert target from tags: [#{tags}]") unless instance
+    instance
+  end
+
+  private
+
+  def whitelist?(_event)
+    true # we collect event by tags, see stream
+  end
+
+  def event_to_hash(event, current_ems_id)
+    event.severity = map_severity(event.severity) # gets into event.full_data
+    event.url = event.tags[TAG_URL]
```

Comment:
again, consider not mutating argument.  I wouldn't expect side effects from a function named `foo_to_hash`.
can do `:full_data => event.to_h.merge(...)`